pkgs/by-name/co/colstr/package.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, fetchFromGitea
+, gcc
+, nix-update-script
+}:
+
+stdenv.mkDerivation rec {
+  pname = "colstr";
+  version = "1.0.0";
+
+  src = fetchFromGitea {
+    domain = "git.sleeping.town";
+    owner = "wonder";
+    repo = "colstr";
+    rev = version;
```

Comment:
why from finalAttrs? With `rec` you can use the version variable already defined.

Reply:
Basically it's a best practice to avoid recursion

Discussion: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes

Reply:
I don't feel like there is a shared opinion in the discussion about what to use now.

Reply:
rec is slightly more practical.
finalAttrs is better structured and in line with the Nixpkgs design principles.